### DISCUSSION TITLE
How to export star data on screen

### DISCUSSION THREAD

**Discussion by selaskar** [Stellarium/stellarium#2424] [Q&A] [answered]
Hello. I am new to Stellarium. I would like to export data about stars on the screen. I use the following script as base: 
https://github.com/raspberrystars/CV-Star-Sensor/blob/master/Stellarium%20Scripts/thesis_4.ssc

It saves consecutive screenshots in a specified region. I modified it to include labels on stars with the following line:
StarMgr.setLabelsAmount(5);

However, I need more detailed data (i.e. position in the image) of either the stars themselves or the labels generated by StarMgr module. The purpose is to train a machine learning module which will later be used for a thesis. Is it possible to export information about objects on the screen to a text file?

I use version 19.0.1 on Windows.

**Comment by gzotti** [comment, marked as answer]
First, please update to at least 0.21.2 to get accurate data.

Ctrl-C copies text data to the clipboard. 

You can retrieve the info in a script via core.getObjectInfo(string objName)  
(https://stellarium.org/doc/0.22/classStelMainScriptAPI.html#aebaa4b5ef785df2227317844e95279d7)

**Comment by selaskar** [comment]
Hey Georg,

Thank you for your answer. I will have 2 follow-up questions, if I may.

First, is there a way to list all the objects (or just one type of objects) visible on the screen at the time?

We found something like StelObjectMgr.listAllModuleObjects(const QString &moduleId, bool inEnglish), but we aren't sure how to use it or if it is what we think.

Second one, does the getObjectInfo() method return any information about the position of the object on the screen?

Thank you very much!

**Reply by gzotti** [reply]
Nothing built-in for you. You could add a data dump routine to the star drawing loop, but this would likely break real-time demands. Or add something that dumps data only when a screenshot is produced. (We are actually doing special renders for "screenshots" to allow other sizes.) Especially for stars, only minimal data are used for drawing, all other data retrieved by the StarMgr only if selected. 
If you center the selected object, it is in the image centre obviously. Else there are no pixel coordinates available after drawing.
